assembly/__tests__/avlTreeContract.ts

```diff
@@ -0,0 +1,46 @@
+import { AVLTree } from "../runtime";
+
```

Comment:
Any top level code is executed at the start before the contract methods are called.  So you can just define it at the top level once.  However, I've found that it's best to import it from another file so that if you wanted to unit test this, the test could also import the tree.  We currently have an issue with unit testing, where even though we wipe away the storage and state of the Mock VM that handles the `env` functions, we don't wipe away the state of the contract.  This means you need access to the same tree object as the "contract" file so you can clear it.  It's fine in your case because since the trees are not in the global scope they won't be used again.

Reply:
I think I understand. Am I correct in thinking that it's imperative that unit tests and contracts use distinct prefixes when creating a collection (e.g. AVLTree)?

Reply:
This would only matter if you were creating the tree is another file and didn't have access to it, e.g. testing contract methods that interact with the tree.  In this case however, you can just do a `beforeEach` and reconstruct the tree.  The issue is that the persistent collections usually have a length or size field, which when not set reads from the storage, and when the storage is refreshed but the local collection isn't it'll have the incorrect size.  So reconstructing the collection before each test ensures that the length is uninitialized and will agree with the storage.

Reply:
Furthermore, since this contract isn't tested with unit tests, it's fine to just use a global variable for the tree.

Reply:
Ah I see! Changes you made make sense.